src/main/java/Doemon.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Scanner;
+
+public class Doemon {
+    /**
+     * Picture of Doemon made using text
+     */
+    private static String logo = "                       _______________\n" +
+            "                      /  --. --.      \\ \n" +
+            "                     /  | '| ' |   \\   \\ \n" +
+            "                    / /  `-O--'     \\   \\ \n" +
+            "                   |.  --  |  --     |   |\n" +
+            "                   |  --   |  --     |   |\n" +
+            "                    \\  (___|_______) /  /\n" +
+            "                     \\              /  /\n" +
+            "                       |== (t) ===|____";
+
+    /**
+     * Introduction string that is printed when Doemon is started.
+     */
+    private static String introStr = "Hello I'm\n" + logo + "\t\t\tDoemon!";
+
+    /**
+     * The string that is printed when Doemon is exited.
+     */
+    private static String exitStr = "I'm going to sleep now...See you again soon!";
+
+    /**
+     * A fixed-sized string array that stores user input.
+     */
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        // Introduction
+        System.out.println(output(introStr));
+
+        new Doemon().start();
+    }
+
+    /**
+     * Starts the Doemon chatbot.
+     */
+    private void start() {
+        // Handling inputs
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            String inputStr = sc.nextLine();
+
+            // Exit the chatbot
+            if (inputStr.equals("bye")) {
+                System.out.println(output(exitStr));
+                break;
+            }
+
+            // List tasks
+            if (inputStr.equals("list")) {
+                int listNum = 1;
+                String listStr = "Here is what's on my bread:\n\t";
+                for (Task task : tasks) {
+                    if (task == null) break;
+                    listStr += listNum++ + "." + task.toString() + "\n\t";
+                }
+                listStr = listNum == 1 ? "You have no tasks!" : listStr.trim();
+                System.out.println(output(listStr));
+                continue;
+            }
+
+            String[] inputArr = inputStr.split(" ");
+
+            // Check for mark/unmark/delete
+            try {
+                if (hasMarkedOrDeleted(inputArr)) continue;
+            } catch (InvalidTaskNumberException itne) {
+                System.out.println(output(itne.toString()));
+            }
+
+            // Add task
+            try {
+                addTask(inputStr, inputArr);
+            } catch (EmptyTaskException ete) {
+                System.out.println(output(ete.toString()));
+            } catch (InvalidTaskException ite) {
+                System.out.println(output(ite.toString()));
+            } catch (MissingArgumentException mae) {
+                System.out.println(output(mae.toString()));
```

Comment:
If I'm not wrong, it is possible to catch multiple exceptions under the same clause.